Resources/Private/src/liststyle/liststyleediting.js

```diff
@@ -150,12 +151,18 @@ function downcastListStyleAttribute() {
 	// @param {String} listStyle
 	// @param {module:engine/view/element~Element} element
 	function setListStyle( writer, listStyle, element ) {
-		Object.keys(getListStyles()).map(configuration => configuration['value']).forEach(className => writer.removeClass(className, element));
+		const listStyles = getListStyles(element.name);
+
+		const className = listStyles[listStyle].value;
+		Object.keys(listStyles).map(listStyle => listStyles[listStyle].value)
+			.forEach(name => {
+				if (name !== className) {
+					writer.removeClass(name, element);
+				}
+			});
 
 		if ( listStyle && listStyle !== DEFAULT_LIST_TYPE ) {
-			writer.setAttribute( 'list-style-type', listStyle, element );
-		} else {
-			writer.removeAttribute('list-style-type', element);
+			writer.addClass(className, element);
```

Comment:
Unfortunately, this doesn't seem to save the list properly.

While the editor displays the list correctly, it splits the list into two when saving to the backend:
UI:
```html
<ol class="neos-list-upper-roman">
  <li>One</li>
  <li>Two</li>
  <li>Three</li>
</ol>
```
Change Payload:
```html
<ol class="neos-list-upper-roman">
  <li>One</li>
</ol>
<ol>
  <li>Two</li>
  <li>Three</li>
</ol>
```

When reloading the backend or publishing after saving, this problem is visible in the ui / frontend. I believe this was exactly the problem I had when I tried using classes (see https://github.com/nlx-lars/list-style/issues/1#issuecomment-1081552539).
Can you reproduce this behavior?

Maybe we should change this back to an attribute and implement classes in a separate PR?